### PR TITLE
watcher: make near archive watcher work

### PR DESCRIPTION
**Pull request by panoel**
This PR addresses the Near explorer endpoint having a low amount of requests per day.  The code is more efficient to conserve these requests.

### CODE REVIEW

File: watcher/src/watchers/__tests__/parseVaa.test.ts [hunk 1]
```diff
@@ -0,0 +1,604 @@
+// import {
```
Comment (evan-gray) [inline watcher/src/watchers/__tests__/parseVaa.test.ts:1]: this file is new, but why is there so much commented out / unused stuff in here?

File: watcher/src/watchers/AlgorandWatcher.ts [hunk 2]
```diff
@@ -124,6 +127,6 @@ export class AlgorandWatcher extends Watcher {
     if (!vaasByBlock[toBlockKey]) {
       vaasByBlock[toBlockKey] = [];
     }
-    return vaasByBlock;
+    return { vaasByBlock, optionalBlockHeight: undefined };
```
Comment (evan-gray) [inline watcher/src/watchers/AlgorandWatcher.ts:130]: simply omit `optionalBlockHeight`

```suggestion
    return { vaasByBlock };
```

File: watcher/src/watchers/AptosWatcher.ts [hunk 3]
```diff
@@ -63,7 +66,7 @@ export class AptosWatcher extends Watcher {
         vaasByBlock[blockKey] = [...(vaasByBlock[blockKey] ?? []), vaaKey];
       })
     );
-    return vaasByBlock;
+    return { vaasByBlock, optionalBlockHeight: undefined };
```
Comment (evan-gray) [inline watcher/src/watchers/AptosWatcher.ts:69]: ditto for this and the rest

File: watcher/src/watchers/Watcher.ts [hunk 4]
```diff
@@ -109,14 +119,35 @@ export class Watcher {
...
-        this.logger.error(e);
+        this.logger.error(`error fetching messages: ${e}`);
         const expoBacko = TIMEOUT * 2 ** retry;
         this.logger.warn(`backing off for ${expoBacko}ms`);
         await sleep(expoBacko);
       }
       if (parentPort) {
         parentPort.postMessage('heartbeat');
       }
+      if (this.chain === 'Berachain') {
```
Comment (evan-gray) [inline watcher/src/watchers/Watcher.ts:130]: maybe there could be a new field on the watcher for these cases like `SLEEP` or something? I think both cases could be covered by the below code.

File: watcher/src/watchers/NearArchiveWatcher.ts [hunk 5]
```diff
@@ -114,21 +119,26 @@ export class NearArchiveWatcher extends Watcher {
...
+      response.lastBlockHeight.toString(),
+      new Date(lastBlockInfo.header.timestamp / 1_000_000).toISOString()
     );
-    if (!vaasByBlock[blockKey]) {
-      vaasByBlock[blockKey] = [];
+    if (!response.vaasByBlock[blockKey]) {
+      response.vaasByBlock[blockKey] = [];
     }
-    return vaasByBlock;
+    return { vaasByBlock: response.vaasByBlock, optionalBlockHeight: response.lastBlockHeight };
```
Comment (evan-gray) [inline watcher/src/watchers/NearArchiveWatcher.ts:141]: couldn't this just return `response`?

**Review by evan-gray — CHANGES_REQUESTED** [review]

File: watcher/src/watchers/__tests__/parseVaa.test.ts (hunk 1 above)
Comment (panoel) [reply watcher/src/watchers/__tests__/parseVaa.test.ts:1]: This file should not have been committed.  I used this to test some SDKv2 functionality.  It has been deleted.

File: watcher/src/watchers/AlgorandWatcher.ts (hunk 2 above)
Comment (panoel) [reply watcher/src/watchers/AlgorandWatcher.ts:130]: Changed.

File: watcher/src/watchers/AptosWatcher.ts (hunk 3 above)
Comment (panoel) [reply watcher/src/watchers/AptosWatcher.ts:69]: Changed.

File: watcher/src/watchers/NearArchiveWatcher.ts (hunk 5 above)
Comment (panoel) [reply watcher/src/watchers/NearArchiveWatcher.ts:141]: Yes, changed.

File: watcher/src/watchers/AlgorandWatcher.ts (hunk 2 above)
Comment (panoel) [reply watcher/src/watchers/AlgorandWatcher.ts:130]: Changed

File: watcher/src/watchers/AptosWatcher.ts (hunk 3 above)
Comment (panoel) [reply watcher/src/watchers/AptosWatcher.ts:69]: Changed

File: watcher/src/watchers/NearArchiveWatcher.ts (hunk 5 above)
Comment (panoel) [reply watcher/src/watchers/NearArchiveWatcher.ts:141]: Yes, changed.

**Review by evan-gray — APPROVED** [review]
I feel like there *might* be a better format for the `watch` loop :thinking: but what is there seems safe.

**Pull request merged**